File: tests/test02.py
import socket
import time
import MySQLdb

def index(req, address="", port=29502, measuredSampleCount=10000, delaySampleCount=1000, db="no", table="data"):
	output = ""
	samples = 1
	
	if db == "yes":
		mysql = MySQLdb.connect("127.0.0.1", "root", "aaa", "rtaixml")
		c = mysql.cursor()
		c.execute("DELETE FROM " + table + " WHERE server = '" + address + "';")
		mysql.commit()

	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
	s.connect((address, int(port)))

	seqLoop = {}
	seqLoopChangeIndicator = {}
	sampleCounts = []
	totalSampleCount = 0
	delayedSampleCount = 0
	clockStarted = False

	while samples:
		
		if not clockStarted and delayedSampleCount >= delaySampleCount:
			executionTime = -time.time()
			clockStarted = True
		
		samples = s.recv(8096)
		
		samplesList = samples.split("\n")
		
		queryValues = []
		for sample in samplesList[:-1]:
			#create mysql string
			value = sample.split(" ")
			
			if len(value) >= 3 and value[0] != "":
				#set sequence loop default values
				if not int(value[0]) in seqLoop:
					seqLoop[int(value[0])] = 0
					seqLoopChangeIndicator[int(value[0])] = False
				
				if int(value[1]) < 32768:
					if seqLoopChangeIndicator[int(value[0])] == True: #increase sequence loop number after sequence number restart
						seqLoop[int(value[0])] += 1
					seqLoopChangeIndicator[int(value[0])] = False; #we do not need to watch for sequence number restart
				else:
					seqLoopChangeIndicator[int(value[0])] = True; #we need to watch for restart
				
				queryValues.append("('" + address + "', '" + str(value[0]) + "', '" + str(seqLoop[int(value[0])]) + "', '" + str(value[1]) + "','" + str(value[2]) + "')")

		sampleCount = len(queryValues)
		if sampleCount > 0:
			if clockStarted:
				if db == "yes":
					c.execute("INSERT INTO " + table + " (server, signalId, seqLoop, seq, value) VALUES " + ",".join(queryValues) + ";")
					mysql.commit()
				sampleCounts.append(sampleCount)
				totalSampleCount += sampleCount
			else:
				delayedSampleCount += sampleCount
		
		if int(measuredSampleCount) and totalSampleCount >= int(measuredSampleCount):
			break
	
	executionTime += time.time()
	s.close()

	#average sample count in one packet
	count = len(sampleCounts) #total numbers in array
	total = 0
	for value in sampleCounts:
		total = total + value #total value of array numbers
	average = float(total)/float(count) #get average value
	output += "\nAverage sample count in one packet: " + str(average).replace(".", ",") + "\n";
	
	#average packet count in time
	output += "\nAverage packet count in one second: " + str(count/executionTime).replace(".", ",") + "\n\n";

	output += "Sample count: " + str(totalSampleCount) + "\n"
	output += "Execution time: " + str(executionTime).replace(".", ",") + " s\n"

	return output

File: tests/test02.php
<?php
header("Content-type: text/plain");

ini_set("memory_limit", "512M");
ini_set("max_execution_time", 3600);

if(!isset($_GET["address"])) $_GET["address"] = "";
if(!isset($_GET["port"])) $_GET["port"] = 29502;
if(!isset($_GET["measuredSampleCount"])) $_GET["measuredSampleCount"] = 10000;
if(!isset($_GET["delaySampleCount"])) $_GET["delaySampleCount"] = 1000;
//if(!isset($_GET["readFunction"])) $_GET["readFunction"] = "fread";
if(!isset($_GET["db"])) $_GET["db"] = "no";
if(!isset($_GET["table"])) $_GET["table"] = "data";

$server = $_GET["address"];

if($_GET["db"] == "yes") {
	mysql_connect("localhost", "root", "aaa");
	mysql_select_db("rtaixml");
	mysql_query("DELETE FROM ".$_GET["table"]." WHERE server = '$server';");
	//mysql_query("TRUNCATE TABLE data;");
	//mysql_query("ALTER TABLE data AUTO_INCREMENT = 1;");
}

$errno = 0;
$errstr = "";

$executionTime = 0;

$fp = fsockopen($_GET["address"], $_GET["port"], $errno, $errstr);
if (!$fp) {
    echo "ERROR: $errno - $errstr<br />\n";
} else {
	$seqLoop = array();
	$seqLoopChangeIndicator = array();
	$sampleCounts = array();
	$totalSampleCount = 0;
	$delayedSampleCount = 0;
	$clockStarted = false;
	
    while($samples = fread($fp, 8096)) {
    	
    	if(!$clockStarted && $delayedSampleCount >= $_GET["delaySampleCount"]) {
    		$executionTime = -microtime(true);
    		$clockStarted = true;
    	}

    	$samplesArray = explode("\n", $samples);
    	array_pop($samplesArray); //remove last (blank) line;
    	
    	$queryValues = array();
    	foreach($samplesArray as $sample) {
    		//create mysql string
    		$value = explode(" ", $sample);
    		if(count($value) >= 3) {
    			//set sequence loop default values
	    		if(!isset($seqLoop[$value[0]])) {
	    			$seqLoop[$value[0]] = 0;
	    			$seqLoopChangeIndicator[$value[0]] = false;
	    		}
	    		
	    		if((int) $value[1] < 32768) {
	    			if($seqLoopChangeIndicator[$value[0]] == true) { //increase sequence loop number after sequence number restart
	    				$seqLoop[$value[0]]++;
	    			}
	    			$seqLoopChangeIndicator[$value[0]] = false; //we do not need to watch for sequence number restart
	    		} else {
	    			$seqLoopChangeIndicator[$value[0]] = true; //we need to watch for restart
	    		}
    			$queryValues[] = "('$server', '".$value[0]."', '".$seqLoop[$value[0]]."', '".$value[1]. "', '".$value[2]."')";
    		}
    	}

    	$sampleCount = count($queryValues); 
    	if($sampleCount > 0) {
    		if($clockStarted) {
	    		if($_GET["db"] == "yes") {
	    			mysql_query("INSERT INTO ".$_GET["table"]." (server, signalId, seqLoop, seq, value) VALUES ".implode(",", $queryValues).";");
	    		}
	    		$sampleCounts[] = $sampleCount;
	    		$totalSampleCount += $sampleCount;
    		} else {
    			$delayedSampleCount += $sampleCount;
    		}
    	}
    	
    	if($_GET["measuredSampleCount"] > 0 && $totalSampleCount >= $_GET["measuredSampleCount"]) break;
    }
    $executionTime += microtime(true);		
	fclose($fp);
	
	//average sample count in one packet
    $count = count($sampleCounts); //total numbers in array
    $total = 0;
    foreach($sampleCounts as $value) {
        $total = $total + $value; // total value of array numbers
    }
    $average = ($total/$count); // get average value
    echo "\nAverage sample count in one packet: ".str_replace(".", ",", $average)."\n";
	
	//average packet count in time
    echo "\nAverage packet count in one second: ".str_replace(".", ",", $count/$executionTime)."\n\n";
    
}

echo "Sample count: $totalSampleCount\n";
echo "Execution time: ".str_replace(".", ",", $executionTime)." s\n";
?>